components/content_settings/core/browser/ad_block_only_mode_content_settings_utils.cc

```diff
@@ -0,0 +1,104 @@
+          ContentSettingsType::BRAVE_HTTPS_UPGRADE,
+      });
+
+  if (!kAdBlockOnlyModeContentSettingsTypes.contains(content_type)) {
+    return false;
+  }
+
+  if (is_off_the_record && kOffTheRecordAwareTypes.contains(content_type)) {
+    return false;
+  }
+
+  return true;
+}
+
+void FillAdBlockOnlyModeRules(OriginValueMap& ad_block_only_mode_rules) {
+  RuleMetaData metadata;
+  metadata.SetExpirationAndLifetime(base::Time(), base::TimeDelta());
+  metadata.set_session_model(mojom::SessionModel::DURABLE);
+
+  base::AutoLock auto_lock(ad_block_only_mode_rules.GetLock());
+  ad_block_only_mode_rules.SetValue(
+      ContentSettingsPattern::Wildcard(), ContentSettingsPattern::Wildcard(),
+      ContentSettingsType::JAVASCRIPT,
+      ContentSettingToValue(CONTENT_SETTING_ALLOW), metadata.Clone());
+
+  ad_block_only_mode_rules.SetValue(
+      ContentSettingsPattern::Wildcard(), ContentSettingsPattern::Wildcard(),
+      ContentSettingsType::COOKIES,
+      ContentSettingToValue(CONTENT_SETTING_ALLOW), metadata.Clone());
+  // ad_block_only_mode_rules.SetValue(
+  //     ContentSettingsPattern::Wildcard(), ContentSettingsPattern::Wildcard(),
+  //     ContentSettingsType::BRAVE_COOKIES,
+  //     ContentSettingToValue(CONTENT_SETTING_ALLOW), metadata.Clone());
+  ad_block_only_mode_rules.SetValue(
+      ContentSettingsPattern::Wildcard(), ContentSettingsPattern::Wildcard(),
+      ContentSettingsType::BRAVE_REFERRERS,
+      ContentSettingToValue(CONTENT_SETTING_ALLOW), metadata.Clone());
+
+  ad_block_only_mode_rules.SetValue(
+      ContentSettingsPattern::Wildcard(), ContentSettingsPattern::Wildcard(),
+      ContentSettingsType::BRAVE_ADS,
+      ContentSettingToValue(CONTENT_SETTING_BLOCK), metadata.Clone());
+  ad_block_only_mode_rules.SetValue(
+      ContentSettingsPattern::Wildcard(), ContentSettingsPattern::Wildcard(),
+      ContentSettingsType::BRAVE_TRACKERS,
+      ContentSettingToValue(CONTENT_SETTING_ALLOW), metadata.Clone());
+  ad_block_only_mode_rules.SetValue(
+      ContentSettingsPattern::Wildcard(), ContentSettingsPattern::Wildcard(),
+      ContentSettingsType::BRAVE_COSMETIC_FILTERING,
+      ContentSettingToValue(CONTENT_SETTING_ALLOW), metadata.Clone());
+
+  // ad_block_only_mode_rules.SetValue(
+  //     ContentSettingsPattern::Wildcard(), ContentSettingsPattern::Wildcard(),
+  //     ContentSettingsType::BRAVE_FINGERPRINTING_V2,
+  //     ContentSettingToValue(CONTENT_SETTING_ALLOW), metadata.Clone());
+
+  ad_block_only_mode_rules.SetValue(
+      ContentSettingsPattern::Wildcard(), ContentSettingsPattern::Wildcard(),
+      ContentSettingsType::BRAVE_REMEMBER_1P_STORAGE,
+      ContentSettingToValue(CONTENT_SETTING_ALLOW), metadata.Clone());
+
+  ad_block_only_mode_rules.SetValue(
+      ContentSettingsPattern::Wildcard(), ContentSettingsPattern::Wildcard(),
+      ContentSettingsType::BRAVE_HTTPS_UPGRADE,
+      ContentSettingToValue(CONTENT_SETTING_ASK), metadata.Clone());
```

Comment:
This setting turns off all HTTPS Upgrades, which means it will be worse than Chrome's default setting.